### DISCUSSION TITLE
404 Issue

### DISCUSSION THREAD

**Discussion by tsrCodes** [pocketbase/pocketbase#2140] [Q&A]
Hey there , Just configure the pocket base instance in ubuntu server with help of reverse Proxy and adding the domain to it , yes its works fine, the problem is when i need to auth the admin using js-sdk it returns 404 error  (`/api/admins/auth-with-password`) , 

Yes everything works fine in admin, URL works well, yes we can access the admin, can create a new admin , can create a collection from adminpanel

```
data: { code: 404, message: 'Not Found.', data: {} },
```

**Comment by ganigeorgiev** [comment]
Admins are not auth collection. If you want to authenticate as admin, you have to use the [`pb.admins` service](https://github.com/pocketbase/js-sdk#adminservice):

```js
await pb.admins.authWithPassword("test@example.com", "1234567890")
```

**Reply by tsrCodes** [reply]
Take a look at Example , did the samething :)

```js
    try {
      await locals.pb.admins.authWithPassword('tsrcodes@gmail.com', 'UzSY8RRfkk72kHK')
    } catch (error) { 
      console.log(error)
    }
```

Using SvelteKit as FrontEnd , Everything is Configure as per docs

**Reply by ganigeorgiev** [reply]
I'm not able to reproduce it. The Admin UI itself also uses the JS SDK, so if you are able to access the Admin UI, then something else is happening.

Are you sure that you are using the correct PocketBase url when you initialize the SDK?

Could you provide a minimal reproducible repo?

**Reply by tsrCodes** [reply]
Its need to be private Repo for some reason, Can i give access to your user id ?? Would you accept

**Reply by ganigeorgiev** [reply]
You don't have to share your private project. Please try to create a **minimal** reproducible repo OR at least provide a more detailed code sample how you initialize the SDK.

**Reply by tsrCodes** [reply]
Exactly use this one [repo](https://github.com/jianyuan/pocketbase-sveltekit-auth)

**Reply by ganigeorgiev** [reply]
I don't see anywhere in the above repo `pb.admins` to be used.

Please if you can't share a **minimal** reproducible repo that shows the error, provide at least a more detailed code sample showing how you initialize the SDK.

**Reply by tsrCodes** [reply]
I think its the problem with the ubuntu server ,Coz  URL itself returns 404 , Can I share the config file ??

**Comment by tsrCodes** [comment]
Stupid Me !! Problem with the Version, Stick to the Basics :( , Anyways Sorry to annoy @ganigeorgiev 💯

**Reply by devbyVeriTech** [reply]
Sorry I’m replying after a year.
I’m facing this issue right now.

What exactly did you rectify?
The node version?

**Reply by darkwiz666** [reply]
At this point it's probably the "pb.admin" change to "collection('_superusers')" in the recent SDKs, but I'm still getting the same issue that he is.

```javascript
import PocketBase from 'pocketbase'
import { S_UN, S_PW } from '$env/static/private'

const pb = new PocketBase("http://127.0.0.1:8090")

async function authenticate() {
    try {
        // Authenticate with the stored username and password
        const authData = await pb.collection('_superusers').authWithPassword(S_UN, S_PW);
        
        console.log('Authentication successful:', authData);
        return authData;
    } catch (error) {
        console.error('Authentication failed:', error);
        throw error; // Optionally re-throw the error for further handling
    }
}

// Example usage
async function main() {
    try {
        const result = await authenticate();
        console.log('Logged in user:', result.record);
    } catch (error) {
        console.error('Failed to login:', error);
    }
}

// Call the main function to run the authentication
main();
```

Same result:

```json
url: 'http://127.0.0.1:8090/api/collections/_superusers/auth-with-password',
  status: 404,
  response: {
    code: 404,
    message: "The requested resource wasn't found.",
    data: {}
  },
```

And I'm just assuming I don't need to do anything specific (like actually request a record) to get a 200.

**Reply by ganigeorgiev** [reply]
@darkwiz666 Make sure that you are using a compatible PocketBase executable and JS SDK versions.

If you are using PocketBase executable v0.23+, then you need to use JS SDK v0.22+.

If you still have troubles, please open a new Q&A discussion with details about the used JS SDK and PocketBase versions.

**Reply by darkwiz666** [reply]
> @darkwiz666 Make sure that you are using a compatible PocketBase executable and JS SDK versions.
> 
> If you are using PocketBase executable v0.23+, then you need to use JS SDK v0.22+.
> 
> If you still have troubles, please open a new Q&A discussion with details about the used JS SDK and PocketBase versions.

Yuuup I totally forgot to update it. The PC was in storage for over 8 months and I just got back around to picking at an older project. 😗
